src/adapter/src/coord/sequencer.rs

```diff
@@ -1052,47 +1055,65 @@ impl Coordinator {
 
         self.catalog_transact(Some(session), vec![op]).await?;
 
-        self.create_cluster_replica(cluster_id, id).await;
+        self.create_cluster_replicas(&[(cluster_id, id)]).await;
 
         Ok(ExecuteResponse::CreatedClusterReplica)
     }
 
-    async fn create_cluster_replica(&mut self, cluster_id: ClusterId, replica_id: ReplicaId) {
-        let cluster = self.catalog.get_cluster(cluster_id);
-        let replica_config = cluster.replicas_by_id[&replica_id].config.clone();
-        let role = cluster.role();
-
-        let log_source_ids: Vec<_> = replica_config.compute.logging.source_ids().collect();
-        let log_source_collections = replica_config
-            .compute
-            .logging
-            .sources
-            .iter()
-            .map(|(variant, id)| (*id, variant.desc().into()))
-            .collect();
+    async fn create_cluster_replicas(&mut self, replicas: &[(ClusterId, ReplicaId)]) {
```

Comment:
Might be I'm missing something, but it seems this could take some `impl IntoIterator<Item=(ClusterId, ReplicaId)>` and then avoid the `Vec` allocation further up?

Reply:
I think that works!

Reply:
oh rip it doesnt, because some callsites need borrow from `self` to get the list of cluster replicas, which would be borrowed by the iterator, while we also need to mutably borrow from `self` to call `create_cluster_replicas`

Reply:
Thanks for checking!